web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts

```diff
@@ -61,19 +62,23 @@ export function useUrlFiltering(
   }
 
   const [initialParamsState] = useState(initialParams);
+  const [includedResourceMode, setIncludedResourceMode] =
```

Comment:
I don't think we should add more state. The resource mode is kept in user preferences so we already have a state for it. 
Ideally, we should probably refactor `useUserFiltering` to set and read `includedResourceMode` from `useUser`, but for now I'd just pass that variable through the hook:
```
Subject: [PATCH] Do not add another state for `includedResourceMode`
---
Index: web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx b/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx
--- a/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx	(revision 2f3d3ad6ad41576e351d81657db5f2315e344747)
+++ b/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx	(date 1731429193528)
@@ -132,15 +132,18 @@
   const canCreate = teleCtx.storeUser.getTokenAccess().create;
   const [loadClusterError, setLoadClusterError] = useState('');
 
-  const { params, setParams, replaceHistory, pathname } = useUrlFiltering({
-    sort: {
-      fieldName: 'name',
-      dir: 'ASC',
-    },
-    includedResourceMode: availabilityFilter?.mode,
-    pinnedOnly:
-      preferences?.unifiedResourcePreferences?.defaultTab === DefaultTab.PINNED,
-  });
+  const { params, setParams, replaceHistory, pathname } = useUrlFiltering(
+    {
+      sort: {
+        fieldName: 'name',
+        dir: 'ASC',
+      },
+      pinnedOnly:
+        preferences?.unifiedResourcePreferences?.defaultTab ===
+        DefaultTab.PINNED,
+    },
+    availabilityFilter?.mode
+  );
 
   const getCurrentClusterPinnedResources = useCallback(
     () => getClusterPinnedResources(clusterId),
Index: web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts b/web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts
--- a/web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts	(revision 2f3d3ad6ad41576e351d81657db5f2315e344747)
+++ b/web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts	(date 1731429193524)
@@ -40,7 +40,8 @@
 }
 
 export function useUrlFiltering(
-  initialParams: Partial<ResourceFilter>
+  initialParams: Partial<Omit<ResourceFilter, 'includedResourceMode '>>,
+  includedResourceMode?: IncludedResourceMode
 ): UrlFilteringState {
   const { search, pathname } = useLocation();
 
@@ -62,8 +63,6 @@
   }
 
   const [initialParamsState] = useState(initialParams);
-  const [includedResourceMode, setIncludedResourceMode] =
-    useState<IncludedResourceMode>(initialParamsState.includedResourceMode);
   const params = useMemo(() => {
     const urlParams = getResourceUrlQueryParams(search);
     return {
@@ -78,7 +77,6 @@
   }, [search, includedResourceMode]);
 
   function setParams(newParams: ResourceFilter) {
-    setIncludedResourceMode(newParams.includedResourceMode);
     replaceHistory(
       encodeUrlQueryParams({
         pathname,
```